compiler/src/iree/compiler/Codegen/LLVMGPU/test/reduction_pipeline.mlir

```diff
@@ -39,7 +39,107 @@ hal.executable.variant @cuda, target = <"cuda", "cuda-nvptx-fb"> {
+//         CHECK:      %[[B:.+]] = vector.broadcast %[[RL:.*]] : f32 to vector<1xf32>
+//         CHECK:      scf.yield %[[B]] : vector<1xf32>
+//         CHECK:    }
+//         CHECK:    %[[R1:.+]] = vector.extract %[[R0]][0] : vector<1xf32>
+//         CHECK:    %[[S0:.+]], %{{.*}} = gpu.shuffle  xor %[[R1]], %[[C1]], %[[C32]] : f32
+//         CHECK:    %[[R2:.+]] = arith.addf %[[R1]], %[[S0]] : f32
+//         CHECK:    %[[S1:.+]], %{{.*}} = gpu.shuffle  xor %[[R2]], %[[C2]], %[[C32]] : f32
+//         CHECK:    %[[R3:.+]] = arith.addf %[[R2]], %[[S1]] : f32
+//         CHECK:    %[[S2:.+]], %{{.*}} = gpu.shuffle  xor %[[R3]], %[[C4]], %[[C32]] : f32
+//         CHECK:    %[[R4:.+]] = arith.addf %[[R3]], %[[S2]] : f32
+//         CHECK:    %[[S3:.+]], %{{.*}} = gpu.shuffle  xor %[[R4]], %[[C8]], %[[C32]] : f32
+//         CHECK:    %[[R5:.+]] = arith.addf %[[R4]], %[[S3]] : f32
+//         CHECK:    %[[S4:.+]], %{{.*}} = gpu.shuffle  xor %[[R5]], %[[C16]], %[[C32]] : f32
+//         CHECK:    %[[R6:.+]] = arith.addf %[[R5]], %[[S4]] : f32
+//         CHECK:    %[[ALLOC:.+]] = memref.alloc() : memref<8xf32, #gpu.address_space<workgroup>>
+//         CHECK:    %[[WID:.+]] = arith.divui %{{.*}}, %{{.*}} : index
+//         CHECK:    %[[LANE_ID:.*]] = arith.remui %[[TID]], %[[C32I]] : index
+//         CHECK:    %[[LANE0:.*]] = arith.cmpi eq, %[[LANE_ID]], %[[C0]] : index
+//         CHECK:    scf.if %[[LANE0]] {
+//         CHECK:      memref.store %[[R6]], %[[ALLOC]][%[[WID]]] : memref<8xf32, #gpu.address_space<workgroup>>
+//         CHECK:    }
+//         CHECK:    gpu.barrier
+//         CHECK:    %[[LANE_ID_IN_BOUNDS:.*]] = arith.minui %[[LANE_ID]]
+//         CHECK:    %[[LOAD_VAL:.+]] = memref.load %[[ALLOC]][%[[LANE_ID_IN_BOUNDS]]] : memref<8xf32, #gpu.address_space<workgroup>>
+//         CHECK:    %[[S5:.+]], %{{.*}} = gpu.shuffle  xor %[[LOAD_VAL]], %[[C1]], %[[C32]] : f32
+//         CHECK:    %[[R7:.+]] = arith.addf %[[LOAD_VAL]], %[[S5]] : f32
+//         CHECK:    %[[S6:.+]], %{{.*}} = gpu.shuffle  xor %[[R7]], %[[C2]], %[[C32]] : f32
+//         CHECK:    %[[R8:.+]] = arith.addf %[[R7]], %[[S6]] : f32
+//         CHECK:    %[[S7:.+]], %{{.*}} = gpu.shuffle  xor %[[R8]], %[[C4]], %[[C32]] : f32
+//         CHECK:    %[[R9:.+]] = arith.addf %[[R8]], %[[S7]] : f32
+//         CHECK:    %[[S9:.+]], %{{.*}} = gpu.shuffle  idx %[[R9]], %[[C0I]], %[[C32]] : f32
+//         CHECK:    %[[R12:.+]] = arith.addf %[[S9]], %[[CF]] : f32
+//         CHECK:    %[[R13:.+]] = vector.broadcast %[[R12]] : f32 to vector<1xf32>
+//         CHECK:    %[[TID0:.+]] = arith.cmpi eq, %[[TID]], %[[C0]] : index
+//         CHECK:    scf.if %[[TID0]] {
+//         CHECK:      vector.transfer_write %[[R13]], %{{.*}}[%{{.*}}] {in_bounds = [true]} : vector<1xf32>, memref<512xf32, #hal.descriptor_type<storage_buffer>>
+//         CHECK:    }
+
+// -----
+
+#pipeline_layout = #hal.pipeline.layout<push_constants = 0, sets = [
```

Comment:
These new tests are just duplications of the CUDA one? Can we actually split the newly added tests into its own files? Like we can have`reduction_pipeline_cuda.mlir` and `reduction_pipeline_rocm.mlir`.